### DISCUSSION TITLE
Question regarding saving scroll position

### DISCUSSION THREAD

**Discussion by VadimZo** [petyosi/react-virtuoso#682] [Q&A]
Hello, is it possible to save exact scroll position with saving index and offset? Currently I'm saving index of post and after this I use ScrollToIndex but scroll position is not like I expect how to possible to add offset to position was exact?

**Comment by williamstein** [comment]
Here is the code that I use to solve this problem:

https://github.com/sagemathinc/cocalc/blob/master/src/packages/frontend/components/virtuoso-scroll-hook.ts

I relicensed it MIT just now (from AGPL) in case you end up wanting to look at or copy/paste from it.  I hope this is useful to somebody, since the code is relatively simple but was (for me) a pain in the butt to write!
